src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/NamingStyles/NamingStyle.cs

```diff
@@ -19,18 +20,34 @@
 
 namespace Microsoft.CodeAnalysis.NamingStyles
 {
-    internal partial struct NamingStyle : IEquatable<NamingStyle>, IObjectWritable
+    [DataContract]
+    internal readonly partial record struct NamingStyle : IObjectWritable
     {
-        public Guid ID { get; }
-        public string Name { get; }
-        public string Prefix { get; }
-        public string Suffix { get; }
-        public string WordSeparator { get; }
-        public Capitalization CapitalizationScheme { get; }
+        [DataMember(Order = 0)]
```

Comment:
❔ Do we not have anything equivalent to `StructLayout.Sequential`? These attributes seem like unnecessary noise.